examples/hello_world/disagg_skeleton/components/kv_router.py

```
@@ -94,5 +97,10 @@ def _cost_function(self, request_prompt):
     # See details at examples/llm/components/kv_router.py
     @endpoint()
     async def check_hit_rate(self, request_prompt: str) -> AsyncIterator[WorkerId]:
-        max_id, max_hit_rate = self._cost_function(request_prompt)
+        metrics = await self.metrics_aggregator.get_metrics()
+        for endpoint in metrics.endpoints:
```

Ruff (0.11.9)

Import `endpoint` from line 25 shadowed by loop variable

(F402)